addons/web_editor/static/src/scss/web_editor.common.scss

```diff
@@ -744,6 +749,9 @@ blockquote {
             h6 {
                 color: o-color($-h6);
             }
+            .h1, .h2, .h3, .h4, .h5, .h6 {
+                color: inherit;
```

Comment:
My guess is that @pierre-pde decided that apply a class to some paragraph to make it bigger and make it stand out is not a good reason to change its color. But I guess it is debatable. In the end... changing any font-size is a weird use case to me so it is difficult to image what the user would really want.

I think any is ok, given the fact that the user can still force a custom font-size/color on top of the class anyway. But indeed, when designing layout for you designers, I think this is not the right choice.

Maybe we can have both and "only hx classes applied with the editor does not come with the color"... but not sure it is better. Probably worth a discussion between PDE and you @stefanorigano :wink:

Reply:
Sorry, I should have explained myself better.
Considering the overall implementation, the edition behaviour is ok.
You click  "h4"  in the **font-size dropdown** and you get the "h4 font-size", perfect 👍 

My confusion is more about the classes that have been chosen to achieve this result since in BS `.h[x]` are used "_for when you want to match the font styling of a heading but cannot use the associated HTML element._" ([ref.](https://getbootstrap.com/docs/5.1/content/typography/#headings))... thus, in the same context (eg. a snippet with `o_cc`), I'd expect that headings tags and classes behave the same (even if currently it's not the case for CC, but still). 

It's probably trivial tho, dunno 🤷🏻‍♂️

Reply:
> .h1 through .h6 classes are also available, for when you want to match the font styling of a heading but cannot use the associated HTML element.

The problem is defining what case that would be. I understand as "use it if you want the font-size of a heading but not on a heading", which is our exact case. Their example below that sentence is actually applying those classes on paragraphs elements. The fact that we color the hx tags / classes is not a bootstrap features, it is ours. So... either we say the color is part of "the font styling of a heading" or not, it's up to us.

I don't know the answer. I am really 50-50 here, I could want both behaviors as an users actually (keeping the color or not). Which I guess would be an argument for "let it keep the color", as the user can always reset it and force it to the text one if he wants too. While a designer using the class in a template won't be able to apply a hx color. But I think more users may find it weird that "increasing the font size" leads to "the color is changing".

Maybe we should indeed not apply hx classes on text via the interface but custom classes only using the font-size... I really don't know anymore. What was your thinking on this @pierre-pde ? Applying the font-size of h4 keeps the margin/line-height/etc you defined for h4 or not? :thinking: What about you @Guillaume-gdi, what did you think during development?

Reply:
@Guillaume-gdi , @qsm-odoo  ,  I would say that "changing the font size" should indeed also change line-height & margin.
It goes along with font-size IMO.
But, I agree that it shouldn't change the color...
 >  But I think more users may find it weird that "increasing the font size" leads to "the color is changing".

I think you are 100% correct

Reply:
> But I think more users may find it weird that "increasing the font size" leads to "the color is changing".

100%, I think we all agree on that 😉
The edition behaviour is ok.

My doubt was about using `.h[x]` classes to change font-size and font-size **only** since, in order to achieve this behaviour, we have to to override the default BS one (`.h1` is always equal to `<h1>`, weight and colors included, no matter what).

Instinctively I would have suggested to use custom classes instead (eg. `o_h4_size`), but It's true that, when inside a CC,  elements behave "inconsistently" to BS regardless, so maybe it's not a big deal(?). 

It's an implementation topic, so if it's ok to @qsm-odoo it's surely ok for me as well 👍 
On our side (designers/coders) we'll have just to keep in mind this new behavior when creating default templates and such (and there are not a lot of occurrences anyways 😉).